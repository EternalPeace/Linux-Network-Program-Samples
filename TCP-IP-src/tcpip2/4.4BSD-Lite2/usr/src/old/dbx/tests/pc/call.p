(*
 * Test program for dbx call command.
 *)

program calltest (input, output);

type String = array [1..4] of char;
var global : integer;

procedure p1 ;
begin
    writeln(output, 'in p1');
    global := 4;
end;

procedure p2 (frommain : integer);
begin
    writeln(output, 'in p2(', frommain:1, ')');
    global := 9;
end;

procedure p3 (s : String; i : integer);
begin
    writeln(output, 'in p3(', s, ', ', i:1, ')');
    global := 10;
end;

procedure startup ;
var
    mainlocal : integer;
begin
    global := 2;
    mainlocal := 19;
    p1;
    p2(mainlocal);
    p3('test', 3);
end;

begin
    startup;
end.
