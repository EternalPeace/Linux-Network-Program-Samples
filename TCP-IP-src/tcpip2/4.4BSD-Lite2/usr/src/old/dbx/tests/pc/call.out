reading symbolic information ...
[1] stop at 33
[1] stopped in startup at line 33
   33       mainlocal := 19;
in p1

p1 returns successfully
in p2(3)

p2 returns successfully
in p3(blah, 3)

p3 returns successfully
mainlocal = 0
fprintf returns successfully
0 
call call
         ^ syntax error
"call" not call-able
too many parameters in call to p1
type mismatch for frommain in call to p2

